src/algebra/pointwise.lean

```diff
@@ -398,6 +398,37 @@ ext $ λ x, ⟨λ hx, let ⟨p, q, ⟨i, hi⟩, ⟨j, hj⟩, hpq⟩ := set.mem_s
 lemma singleton_smul [has_scalar α β] {t : set β} : ({a} : set α) • t = a • t :=
 image2_singleton_left
 
+instance smul_comm_class_set {γ : Type*}
+  [has_scalar α γ] [has_scalar β γ] [smul_comm_class α β γ] :
+  smul_comm_class α (set β) (set γ) :=
+{ smul_comm := λ a T T', begin
+    ext c,
+    simp only [mem_smul_set, mem_smul],
+    split,
+    { rintro ⟨_, ⟨b, c, hb, hc, rfl⟩, rfl⟩,
+      exact ⟨b, _, hb, ⟨c, hc, rfl⟩, (smul_comm a b c).symm⟩, },
+    { rintro ⟨b, _, hb, ⟨c, hc, rfl⟩, rfl⟩,
+      exact ⟨_, ⟨b, c, hb, hc, rfl⟩, smul_comm a b c⟩ }
+  end }
```

Comment:
Golfed:
```suggestion
{ smul_comm := λ a T T',
    by simp only [←image2_smul, ←image_smul, image2_image_right, image_image2, smul_comm] }
```

Reply:
This approach probably gets you the `is_scalar_tower` proof easily too.

Reply:
Can't figure out an easy way to do these proofs. The `rintro; exact` proofs are quite mindless so I'm happy to keep them.